activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -356,7 +356,11 @@ def find_with_associations
 
     def construct_join_dependency(joins = [])
       including = eager_load_values + includes_values
-      ActiveRecord::Associations::JoinDependency.new(@klass, including, joins)
+      finder_methods_hash = {
+        eager_load: eager_load_values,
+        includes: includes_values
+      }
+      ActiveRecord::Associations::JoinDependency.new(@klass, including, joins, finder_methods_hash)
```

Comment:
So we could pass in nil for the `including` param (needs to be passed in to call `initialize` properly), but we could also use the passed in `finder_methods_hash` where `including` would have been used. And this way we would only have to call `eager_load_values` and `includes_values` once each. Not sure how we feel about just passing in nil for `including` though.